### DISCUSSION TITLE
Making SQLAlchemy not create specific tables

### DISCUSSION THREAD

**Discussion by carlbordum** [sqlalchemy/sqlalchemy#9969] [Usage Questions] [answered]
Hi. Thank you for sqlalchemy, it's really been great.

I'm trying to create a materialized view using [`create_materialized_view`](https://sqlalchemy-utils.readthedocs.io/en/latest/view.html#create-materialized-view) from `sqlalchemy-utils`.

The creation of the view is working fine, but I would like a declarative model in SQLAlchemy aswell, so I can query it and all transparently. Is there a way to create a declarative model in SQLAlchemy, but tell SQLAlchemy that `Base.metadata.create_all` should not create it? I tried using `__abstract__ = True`, but with this I cannot access the fields.

**Comment by zzzeek** [comment, marked as answer]
easiest way is to use a different `MetaData` collection for the materialized view object.  create_all() also accepts a list of tables: https://docs.sqlalchemy.org/en/20/core/metadata.html#sqlalchemy.schema.MetaData.create_all but that's the opposite of an "exclude" list.

**Reply by carlbordum** [reply]
Thank you. For anyone with the same issue, [here](https://docs.sqlalchemy.org/en/20/orm/persistence_techniques.html#partitioning-strategies-e-g-multiple-database-backends-per-session) is some documentation on this, and also note that you can use the `metadata` parameter for `create_materialized_view` to pass _another_ (your main?) metadata than the pseudo-one you just created.

**Reply by carlbordum** [reply]
Do you think my issue is common enough to warrant an API addition?

Perhaps an `__external__ = True` on declarative tables could have this effect, or an exclude list to `MetaData.create_all`?
